src/media/js/addon/reducers/addon.js

```diff
@@ -73,6 +73,17 @@ export default function addonReducer(state=initialState, action) {
       return newState;
     }
 
+    case dashboardActions.DELETE_OK: {
+      /*
+        Mark add-on as deleted.
+
+        payload (string) -- addonSlug
+      */
+      const newState = _.cloneDeep(state);
+      newState.addons[action.payload].deleted = true;
```

Comment:
this makes it easy for the add-on detail page to display a "Your add-on was deleted" message, else the page will render something broken